doc/1-intro/glossary.md

```diff
@@ -0,0 +1,145 @@
+# glossary of common terms
+
+**2-op**, **3-op**, **4-op**...: the number of FM operators used to generate a sound. more operators allow for more complex sounds.
+
+**ADPCM**: adaptive differential pulse code modulation. this is a variety of DPCM with a more complex method of storing the amplitude differences.
+
+**ADSR**: attack, decay, sustain, release. these are the four necessary values for a basic volume envelope.
+
+**algorithm**: the way in which the operators in an FM instrument interact.
+- when two operators connect to the same point, their sounds are added together.
+- when two operators are connected left to right, the left is the modulator and the right is the carrier sound that is modified.
+
+**bitbang**: to achieve PCM sound by sending a rapid stream of volume commands to a non-PCM channel.
+
+**BRR**: a type of lossy ADPCM used by the SNES. it has a fixed compression ratio; groups of 32 bytes (16 samples) are encoded in 9 bytes each.
+- usually stored in .BRR files.
+
+**clipping**: when a sample or playback stream exceeds the maximum or minimum values. this can cause audible distortion.
+- this often occurs when a sample is amplified too much.
+- it can also occur during playback if too much sound is being added together at once. in some cases the mixer can be used to reduce the volume. if this doesn't work, the clipping is caused within the chip's own mixing, and the only solution is to reduce the volumes of the notes being played.
+
+**clock rate**: the timing at which a chip operates, expressed as cycles per second (Hz).
+- changing this may change aspects of how some chips work, most notably pitch.
+- some chips cannot operate at anything other than their designed clock rate.
+
+**cursor**: the marker of input focus. anything typed will happen at the cursor's location.
+- _Furnace:_ this always refers to the pattern view except when in a text entry box.
+
+**DAC**: digital analog converter. this converts a digital representation of sound into actual output.
+
+**.DMF**: a DefleMask song file.
+- _Furnace:_ DFM files may be read, and compatibility flags will be set to make them play as accurately as possible, but there may still be glitches.
+- _Furnace:_ DFM files may be saved, but full compatibility isn't guaranteed and many features will be missing. this isn't recommended unless absolutely necessary.
+
+**.DMP**: a DefleMask instrument file.
+
+**.DMW**: a DefleMask wavetable file.
+
+**DPCM**: differential/delta pulse code modulation. this is a variety of PCM that stores each amplitude as its difference from the previous.
+
+**duty**: in a pulse wave, this represents the ratio of the "on" part of the wave.
```

Comment:
The proper term is "duty cycle", but in most cases I abbreviate it to "duty".

Nah, the true term is "pulse width". Nintendo used "duty cycle" since the selection of pulse widths on NES and Game Boy is small.